students/urtnowski_daniel/lesson_10_organizing_files/deleting_unneeded_files.py

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env python3
+
+"""
+deleting_unneeded_files.py: a practice project "Deleting Unneeded Files" from:
+https://automatetheboringstuff.com/chapter9/
+
+The program walks through a folder tree and searches for exceptionally large
+files or folders (size > 100 MB).
+
+Usage: ./deleting_unneeded_files.py <folder_path>
+
+"""
+
+__author__ = "Daniel Urtnowski"
+__version__ = "0.1"
+
+
+import sys
+import os
+
+
+BYTES_IN_MEGABYTE = 1024 * 1024
+FILE_SIZE_MAX = 100 * BYTES_IN_MEGABYTE
+
+
+def get_args():
+
+    args_cnt = len(sys.argv)
+
+    if args_cnt != 2:
+        print('Usage: ./deleting_unneeded_files.py <path>')
+        return None
+
+    else:
+        path = sys.argv[1]
+
+        if not os.path.exists(path) or not os.path.isdir(path):
+            print("Error! The path: {} is incorrect!".format(path))
+            return None
+
+        return path
+
+
+def main():
+
+    path = get_args()
+    if path is not None:
```

Comment:
Żeby uniknąć tzw. pyramid of doom tworzącej się przy wielokrotnym zagnieżdżaniu bloków kodu można sprawdzić wszystkie warunki na początku funkcji ale odwrócić ich logikę i wyjść z funkcji jeśli nie są spełnione:
```
if path is None:
    return
for curr_.....
```
W tym przypadku zaoszczędzimy tylko 1 poziom zagnieżdżenia, ale moim zdaniem i tak warto. Dodatkowo czytający od razu widzi, że jeżeli warunek nie jest spełniony to wychodzimy z funkcji i nie trzeba szukać `else` na końcu tego `if`a